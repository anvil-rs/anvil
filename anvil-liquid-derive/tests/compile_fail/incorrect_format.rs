// This file tests the error when a template has an incorrectly formatted template attribute
// Expected error: "Expected template attribute to be in the form #[template(\"template_name.ext\")] or #[template(path = \"...\", ...)]"

use anvil_liquid_derive::Template;
use serde::Serialize;

#[derive(Serialize, Template)]
#[template(123)]  // Using a number instead of a string literal
struct IncorrectFormatTemplate {
    value: String,
}

fn main() {
    // This will never run, we're only testing the compile-time error
}
