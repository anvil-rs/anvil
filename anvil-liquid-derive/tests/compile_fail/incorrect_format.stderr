error: unexpected literal in nested attribute, expected ident
 --> tests/compile_fail/incorrect_format.rs
  |
  | #[template(123)]  // Using a number instead of a string literal
  |            ^^^
